Add the item glossary

exoRift opened the pull request:
Add a glossary that will be populated by a script that automatically generates entries based on data from the latest release

At the moment, the glossary will list
- Items
- Blocks
- Mobs
- Brew Effects
- Rites
- Spells

exoRift committed:
Update Font API URL

exoRift committed:
Add constant navbar height

exoRift committed:
Add skeleton for glossary index

MsRandom reviewed (approved):
Pretty clean so far.

exoRift committed:
Add blowup skeleton

exoRift committed:
Implement vanilla item fetching

exoRift committed:
Complete crafting table grid for recipes

exoRift committed:
Glossary mobile support

exoRift committed:
Better mobile support and navigation systems

exoRift committed:
Implement search bar to glossary and hint at mobile index

exoRift committed:
Implement furnace crafting GUI for glossary

exoRift commented:
@MsRandom Stats desired to be served via API

General:
- Name
- Description
- ID
- Recipe (Make sure it also says how many of the item are created by the recipe, like 3 potions from the kettle, for example)
- Recipe type (shaped crafting table, furnace, kettle, etc.)

Items:
- Name
- Description
- ID
- Durability
- Damage
- Type (item, tool, armor, etc.)
- Capacity level
- Power level
- Duration level
- Dispersal type

Blocks:
- Name
- Description
- ID
- Recipe?
- Solid/Transparent
- Type (building, machine)

Mobs:
- Name
- Description
- ID
- Hitpoints
- Damage
- Friendly/Hostile/Boss

Brew Effects:
- Name
- Description
- ID
- Recipe
- Capacity
- Cannot be paired with
- Max stats

Rites:
- Name
- Description
- ID
- Recipe
- Circle required (part of recipe)
- Infusion required
- Coven witches needed

Spells:
- Name
- Description
- ID
- Symbol
- Unlocked with: ?
- Infusion required

Other Requests
- Cache a list of items whos recipe contain an item, per item (That way, I can list which items require an ingredient to craft them on that ingredient's page)

MsRandom commented:
Most of this does appear to be doable.

exoRift committed:
Add shuffling for shapeless crafting

exoRift committed:
Add kettle GUI to glossary

exoRift committed:
Redesign kettle GUI in glossary

exoRift committed:
Change site version and add support for multiple recipes per item

exoRift committed:
Add support for stack sizes

exoRift commented:
TODO:

Add support for
- Category determination (dynamic variables)
- NBT
- Byproducts
- Recipes being served standalone
- Multi-product recipes
- Stack sizes on products
- Hot-link to the machine needed to craft an item on the title of the crafting grid

exoRift committed:
Add spinning wheel GUI

- Add fallback for unknown items
- Fix issue where stack sizes would only display in the crafting bench

exoRift committed:
Split entries and recipes and support multiple products

exoRift committed:
Add Witches' Oven GUI

exoRift committed:
Support uncertain products

exoRift committed:
Add distillery GUI

exoRift committed:
Minor changes

- Navbar no longer creates scrollbars on expansion
- Item descriptions now segment into paragraphs

exoRift committed:
Add compendium page and its index

exoRift committed:
Adjust navbar to be more stable on mobile

exoRift committed:
Trim and polish root HTML file and manifest

exoRift committed:
Use Google's material icons as opposed to a local library

exoRift committed:
Syntactic changes

exoRift committed:
Update home backvid to not scale with page resizing

exoRift committed:
Stop IntersectionObserver on Navbar unmount

exoRift committed:
Add implicit Discord authorization to version upload

exoRift committed:
Fix glossary scrolling

exoRift committed:
Fix crafting grid tooltips

exoRift committed:
Switch from CRA to Vite

exoRift committed:
Add autoprefixer to Vite config

exoRift committed:
Switch Vite export dir from dist to build

exoRift committed:
Trim dependencies

MsRandom committed:
Merge

MsRandom committed:
Fix conflicts
